src/CMakeLists.txt

```diff
@@ -33,16 +32,22 @@ IF (ENABLE_CONDA)
     SET (libsolv_SRCS ${libsolv_SRCS} conda.c)
 ENDIF (ENABLE_CONDA)
 
+IF (NOT WITH_OPENSSL)
+  SET (libsolv_SRCS ${libsolv_SRCS} md5.c sha1.c sha2.c)
+ENDIF (NOT WITH_OPENSSL)
 
 SET (CMAKE_C_FLAGS "${CMAKE_C_FLAGS} -fPIC")
 IF (HAVE_LINKER_VERSION_SCRIPT)
 SET (CMAKE_SHARED_LINKER_FLAGS "${CMAKE_SHARED_LINKER_FLAGS} ${LINK_FLAGS} -Wl,--version-script=${CMAKE_SOURCE_DIR}/src/libsolv.ver")
 ENDIF (HAVE_LINKER_VERSION_SCRIPT)
 
 IF (DISABLE_SHARED)
-ADD_LIBRARY (libsolv STATIC ${libsolv_SRCS})
+  ADD_LIBRARY (libsolv STATIC ${libsolv_SRCS})
```

Comment:
this one should be linked with openssl if enabled too (IIRC OpenSSL supports static linking).

Reply:
@ignatenkobrain I done change. But I am not sure if I understand your request.